src/main/scala/cromwell/engine/workflow/WorkflowActor.scala

```diff
@@ -149,9 +166,12 @@ case class WorkflowActor(workflow: WorkflowDescriptor,
       stay()
     case Event(ExecutionStoreCreated(startMode), NoFailureMessage) =>
       if (startMode == Restart) {
-        symbolsMarkdownTable foreach { table => log.info(s"Initial symbols:\n\n$table") }
-        executionsMarkdownTable foreach { table => log.info(s"Initial executions:\n\n$table") } 
+        dumpTables()
+        stay()
+      } else {
+        startRunnableCalls()
       }
+    case Event(TablesDumped, NoFailureMessage) =>
       startRunnableCalls()
```

Comment:
This should be the async equivalent of the original code.  If in restart mode dump all the tables, then call `startRunnableCalls`.  Before that all happened synchronously, now there's an `onComplete` handler on the `dumpTables` `Future` that sends notifies that the tables have been dumped and the runnable calls can be started.